crates/dbsp/src/circuit/runtime.rs

```diff
@@ -784,8 +790,12 @@ impl Runtime {
         })
     }
 
-    pub fn worker_panic_info(&self, worker: usize) -> Option<WorkerPanicInfo> {
-        if let Ok(guard) = self.inner().panic_info[worker].read() {
+    pub fn worker_panic_info(
+        &self,
+        worker: usize,
+        thread_type: ThreadType,
+    ) -> Option<WorkerPanicInfo> {
+        if let Ok(guard) = self.inner().panic_info[worker][thread_type].read() {
```

Comment:
I can't comment on unmodified lines but it might be good to log a warn!() in the else clause here (can't get panic info due to read-poisoned lock)

Reply:
I added a commit to do that.